### ISSUE TITLE
[BUG] First of the month is always sunday

### ISSUE THREAD

**Issue by Lachrimae93** [vigoren/foundryvtt-simple-calendar#643]
**Describe the bug**
Every month the start of it (1st of the month) the day is always on sunday regardless if the previous month ended on a different day. If the month ended with wednesday, the 1st of the next month is still set to sunday..

**To Reproduce**
1. go to the gregorian calendar (or any setting) and pick a date. 
2. set the date to whatever you want it to be
3. click save configuration and it'll rearrange the dates to make the first of each month always a sunday

**Expected behavior**
Well, it should be that the start of the month is on the day after the end of the last month.

**Comment by miromonti** [comment]
Reporting the same here. I thought the issue was game calendar year 1933 (pre-1970 reference year) messing with something. But now, maybe something else instead?

Screen captures of five months:
![1933 08](https://github.com/user-attachments/assets/a28aabfb-6ab8-4174-a941-e555c2b2f6cc)
![1933 09](https://github.com/user-attachments/assets/278f8aa9-6fbe-4101-8da9-30b0253985c0)
![1933 10](https://github.com/user-attachments/assets/c7031cff-aaf9-48a9-adfe-f50c597aed0a)
![1933 11](https://github.com/user-attachments/assets/8e4607e7-80c1-4bd4-9bf4-bc375df4fb06)
![1933 12](https://github.com/user-attachments/assets/1d0c4b82-ca19-4859-8b17-74c7a145bfb5)

Notice the moon phases, seasons, and expected number of days per month are tracking correctly. (Moon phases presumed correct.) 

For the record:
Simple Calendar 2.4.18
Call of Cthulhu 7e 7.1
FVTT 12.331

**Comment by LeKopic** [comment]
Have you tried adjusting the calendar manually? I am fairly certain it set the months to start on Sundays for some reason.

**Comment by miromonti** [comment]
> Have you tried adjusting the calendar manually? I am fairly certain it set the months to start on Sundays for some reason.

Don't know about that setting (never touched it before), but found "Starting Week Day" in the "Weekday Settings" tab. Is that it? If that's correct, it was set for Thursday, which—on it's own—doesn't seem influential:

![Screenshot 2024-10-28 at 3 12 38 PM](https://github.com/user-attachments/assets/719724a1-34f4-40d0-9572-6c3641bdf27d)

For fun, I set it to Tuesday. The calendar was unfazed and stayed stuck to Sunday every month.

For the heck of it, I chose a calendar that has nothing to do with our planet or time:

![Screenshot 2024-10-28 at 3 19 18 PM](https://github.com/user-attachments/assets/2bf6ee1a-9593-400e-b22d-51038bd04ebe)

Does the Warhammer calendar use Sunday, Monday, ... Saturday?

![Screenshot 2024-10-28 at 3 19 55 PM](https://github.com/user-attachments/assets/0be3dd53-030f-4873-bb58-17c96e8b953f)

Finally, I employed Find the Culprit. With Simple Calendar and a handful of library modules, the calendar remained stuck with every month starting on Sunday. FtC's assessment says:

![Screenshot 2024-10-28 at 3 17 05 PM](https://github.com/user-attachments/assets/18909b73-01f3-4c6c-bcc9-50af0ee1477a)

Is there a "reset to factory" button for Simple Calendar? 😄 (Preferably affecting just the active game world and not across all worlds.)

I hope the above is useful.

**Comment by Lachrimae93** [comment]
> Have you tried adjusting the calendar manually? I am fairly certain it set the months to start on Sundays for some reason.

There doesn't seem to be a setting to do this. All there is is setting the day of the starting year. It defaults to 1970 which starts Thursday (screenshots are posted above here by someone else). It doesn't fix anything. Resetting the calendar by picking the gregorian format seems to work but the moment you "save configuration" it screws up. 

Of course, you can skip clicking save configuration but it doesn't actually fix the problem when I've got a custom calendar for my game and i can't figure out how to stop it from resetting.

**Comment by Sandvich** [comment]
I noticed this issue the other day, and after a bit of debugging it seems that setting the year also causes the same problem. The forward/back one month buttons are fine, but entering a year just breaks it.
For now, I manually clicked through to July 2075 (not fun!) and so that's fixed it, but I want to put in sunrise/sunset timings as those are useful for a number of my group, so I'm looking forward to a fix! Happy to help with testing or similar, but sadly I don't speak javascript so I can't help with debugging or coding so much.

**Comment by miromonti** [comment]
I would like to try that but Simple Calendar seems broken indeed. Whether I change "1933" back to "2024" or "1970" or just remove the year (in which case, current year defaults to "0"), the calendars remain stuck with 1st of the month starting on Sunday.

I deselected Simple Calendar and restarted, then reselected the module and restarted. The default year is "2024", but the calendars remain stuck with 1st of the month on Sunday.

Seriously, is there a reset button or equivalent thereof?

**Comment by miromonti** [comment]
I found a way to recover:

1. Create new game world with same game system. 
2. Load Simple Calendar.
3. Check that it's behaving correctly. (It was.)
4. _Export this calendar's settings._
5. Return to game with faulty calendar.
6. _Import that calendar's settings._

Simple Calendar returns to nominal.

Then a simple 1092 month-clicks into the past and the calendar is ready for my game's 1933.

**Comment by Lachrimae93** [comment]
> I found a way to recover:
> 
> 1. Create new game world with same game system.
> 2. Load Simple Calendar.
> 3. Check that it's behaving correctly. (It was.)
> 4. _Export this calendar's settings._
> 5. Return to game with faulty calendar.
> 6. _Import that calendar's settings._
> 
> Simple Calendar returns to nominal.
> 
> Then a simple 1092 month-clicks into the past and the calendar is ready for my game's 1933.

This doesn't help my one issue where I made an entire custom calendar for my world and can't do anything about the fact that each month now starts at my equivalent of sunday.

**Comment by Sandvich** [comment]
@miromonti 
> I found a way to recover:
> 
> -snipped-
>
> Simple Calendar returns to nominal.

@Lachrimae93 mentioned a much simpler method above. Just select the gregorian calendar again, go through the default settings and leave it all as is. Once you get to the current date, you can set the date via the month buttons without screwing it up, but make sure you don't click the save configuration button at the bottom.

@Lachrimae93 
> This doesn't help my one issue where I made an entire custom calendar for my world and can't do anything about the fact that each month now starts at my equivalent of sunday.

Yeah, at the end of the day this is a workaround that only helps with people using a builtin calendar (assuming it works for non-gregorian calendars that are built in?). You need an actual fix, which requires dev time. No idea when that's coming, as I'm not part of the team.

**Comment by miromonti** [comment]
@Lachrimae93 Indeed. My method is a reset and of no use to you who want to save your calendar. Here's hoping a proper fix or some revelation to patch comes around sooner than later.

@Sandvich I appreciate the mention. Tried it and it didn't work for me. I essentially A-B tested at least a dozen methods before I gave up. Seems my calendar could not be "tricked". Now, with the exported settings, I have the reset button I've been pining for—essentially two clicks.

That all said, others caught in this situation should try the methods above before resorting to reset, especially if your calendar is custom.

**Comment by Lachrimae93** [comment]
@vigoren will there be any plans to fix this at any point in the future?

**Comment by Einkatzi** [comment]
I have the same problem.
Fun fact: in the configuration json for each calendar, startingWeekday may be a number or null - if it's null (as it is in most predefined calendars), the months start with the correct (variable) weekday.
But whenever the configuration screen is opened, every month is shown to start with weekday 1, and when saving configuration, the configuration json is overwritten with startingWeekday:1.
So I think the culprit must be somewhere in https://github.com/vigoren/foundryvtt-simple-calendar/blob/f09bb9697df4e1c58d8758726d8a35089bc77644/src/classes/applications/configuration-app.ts - but my JS/TS knowledge is too shallow to tell where it's going wrong.
Obviously, whenever the json says startingWeekday:null, the configuration screen turns this into startingWeekday:1 and overwrites anything that was correct.

A possible workaround would be to configure the calendar so everything is perfect and nothing needs to be changed afterward, then export it, replace every occurence of startingWeekday:1 with startingWeekday:null, save the file and re-import it. Now the weekdays are right, but every change within the configuration screen will overwrite this.
